File: ConexaoVagas App/models/habilidade.ts
import { HabilidadeCandidato } from "./habilidadeCandidato";
import { HabilidadeVaga } from "./habilidadeVaga";
import { Usuario } from "./usuario";

export class Habilidade {
    idHabilidade?: number;
    idCriadoPor?: number | null;
    nomeHabilidade?: string;
    idCriadoPorNavigation?: Usuario = undefined;
    habilidadeCandidato?: HabilidadeCandidato[] = [];
    habilidadeVaga?: HabilidadeVaga[] = [];
}

File: ConexaoVagas API/Domains/Habilidade.cs
using System;
using System.Collections.Generic;

namespace ConexaoVagasAPI.Domains
{
    public partial class Habilidade
    {
        public Habilidade()
        {
            HabilidadeCandidato = new HashSet<HabilidadeCandidato>();
            HabilidadeVaga = new HashSet<HabilidadeVaga>();
        }

        public int IdHabilidade { get; set; }
        public int? IdCriadoPor { get; set; }
        public string NomeHabilidade { get; set; }

        public Usuario IdCriadoPorNavigation { get; set; }
        public ICollection<HabilidadeCandidato> HabilidadeCandidato { get; set; }
        public ICollection<HabilidadeVaga> HabilidadeVaga { get; set; }
    }
}
